firmware/gameport-adapter/CH-f16-combat-stick.h

```diff
@@ -0,0 +1,88 @@
+// This file is part of Necroware's GamePort adapter firmware.
+// Copyright (C) 2021 Necroware
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+//
+// CH F16 Combat Stick from 1995
+// Made by Stefan Geissler
+//  3 axes
+// 10 buttons
+//  1 Hat with 4 positions
+// The 4th axis (index 2) is ignored, because there is a big jitter
+// and it reacts on movement of the other axes too. So you can not
+// assign the axes to a function in your game.
+
+#pragma once
+
+#include "AnalogJoystick.h"
+#include "Joystick.h"
+
+class CHF16CombatStick : public Joystick {
+public:
+  const Description &getDescription() const override {
+    static const Description description{"CH F16 Combat Stick", 3, 10, 1};
+    return description;
+  }
+
+  const State &getState() const override {
+    return m_state;
+  }
+
+  bool init() override {  
+    return true;
+  }
+
+
+  bool update() override {
+
+    // Unfortunately I had no real CHFlighstickPro joystick to test, but
```

Comment:
This complete comment is been copy pasted from Sidewinder 3D implementation. I guess, it can be removed, because you probably have tested your CH F16 Combat Stick. At least it doesn't make a lot of sense as is.